modules/network/src/main/java/org/apache/ignite/internal/network/direct/DirectMessageReader.java

```diff
@@ -371,6 +383,20 @@ public <C extends Collection<?>> C readCollection(String name, MessageCollection
         return col;
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public <C extends List<?>> C readList(String name, MessageCollectionItemType itemType) {
+        DirectByteBufferStream stream = state.item().stream;
+
+        Collection<?> col = stream.readCollection(itemType, this);
+
+        lastRead = stream.lastFinished();
+
+        assert col instanceof List;
```

Comment:
Hm, so all our collections are always lists?

Reply:
You told me that, when we read them, we always create a list